File: lms-app/scripts/generate-assets.mjs
/**
 * generate-assets.mjs
 * Resizes and copies generated brand assets into the correct Expo asset paths.
 * Run: node scripts/generate-assets.mjs
 */

import { createCanvas, loadImage } from 'canvas';
import { writeFileSync, mkdirSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const ROOT = join(__dirname, '..');
const ASSETS_IMG = join(ROOT, 'assets', 'images');

// Source images (generated)
const SRC = {
  icon:         '/Users/arup/.gemini/antigravity/brain/3bbc70a0-eb73-496d-9c24-538538c5d404/lms_app_icon_1776149574632.png',
  favicon:      '/Users/arup/.gemini/antigravity/brain/3bbc70a0-eb73-496d-9c24-538538c5d404/lms_favicon_1776149750274.png',
  adaptive:     '/Users/arup/.gemini/antigravity/brain/3bbc70a0-eb73-496d-9c24-538538c5d404/lms_adaptive_icon_1776149810581.png',
  splash:       '/Users/arup/.gemini/antigravity/brain/3bbc70a0-eb73-496d-9c24-538538c5d404/lms_splash_icon_1776149846942.png',
};

// Target sizes
const TARGETS = [
  { src: 'icon',     dest: 'icon.png',          size: 1024 },
  { src: 'favicon',  dest: 'favicon.png',        size: 48   },
  { src: 'adaptive', dest: 'adaptive-icon.png',  size: 1024 },
  { src: 'splash',   dest: 'splash-icon.png',    size: 1024 },
];

async function resizeAndSave(srcPath, destPath, size) {
  const img = await loadImage(srcPath);
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  ctx.drawImage(img, 0, 0, size, size);
  const buf = canvas.toBuffer('image/png');
  writeFileSync(destPath, buf);
  console.log(`  ✓ ${destPath} (${size}x${size})`);
}

mkdirSync(ASSETS_IMG, { recursive: true });

console.log('\n🎨 Mini LMS — Asset Pipeline\n');
for (const t of TARGETS) {
  await resizeAndSave(SRC[t.src], join(ASSETS_IMG, t.dest), t.size);
}
console.log('\n✅ All assets written to assets/images/\n');

File: lms-app/scripts/generate_assets.py
#!/usr/bin/env python3
"""
Mini LMS — Asset Pipeline
Resizes and copies generated brand assets into the correct Expo asset paths.
"""
from PIL import Image
import os, shutil

BRAIN = "/Users/arup/.gemini/antigravity/brain/3bbc70a0-eb73-496d-9c24-538538c5d404"
DEST  = "/Users/arup/Desktop/mini-LMS/lms-app/assets/images"

TARGETS = [
    # (source_file, dest_file, size)
    ("lms_app_icon_1776149574632.png",      "icon.png",           1024),
    ("lms_favicon_1776149750274.png",       "favicon.png",          48),
    ("lms_adaptive_icon_1776149810581.png", "adaptive-icon.png",  1024),
    ("lms_splash_icon_1776149846942.png",   "splash-icon.png",    1024),
]

os.makedirs(DEST, exist_ok=True)

print("\n🎨  Mini LMS — Asset Pipeline\n")
for src_name, dst_name, size in TARGETS:
    src_path = os.path.join(BRAIN, src_name)
    dst_path = os.path.join(DEST, dst_name)
    img = Image.open(src_path).convert("RGBA")
    img = img.resize((size, size), Image.LANCZOS)
    img.save(dst_path, "PNG", optimize=True)
    kb = os.path.getsize(dst_path) / 1024
    print(f"  ✓  {dst_name:30s}  ({size}x{size})  {kb:.1f} KB")

print("\n✅  All assets written to assets/images/\n")
